src/script/interpreter.h

```diff
@@ -162,6 +162,18 @@ class BaseSignatureChecker
     virtual ~BaseSignatureChecker() {}
 };
 
+/** A general purpose signature checker. Note that this lacks assumptions about the `sighash`, as the preimage is not known to the signer. */
```

Comment:
This should emphasize that the issue about not knowing the hash preimage is for the verifier, not the signer.  Perhaps:
```
/** A low level signature checker.  Note that the caller must verify that the hash passed in relates to a known message (unlike for the other signature checkers). */

Reply:
Sorry for delay. Yes, that looks great, thanks. Adopting.